npm/vite-dev-server/src/resolveConfig.ts

```diff
@@ -79,12 +80,11 @@ function makeCypressViteConfig (config: ViteDevServerConfig, vite: Vite): Inline
     paths: [projectRoot],
   })))
 
-  return {
+  const viteConfig: InlineConfig = {
```

Comment:
I was going to suggest using the `satisfies` operator so the `!` assertions wouldn't be necessary below, but apparently our linter config doesn't know about it yet 😢 .